src/main/java/com/znsio/rpi/properties/Config.java

```diff
@@ -0,0 +1,43 @@
+package com.znsio.rpi.properties;
+
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.apache.log4j.Logger;
+
+import javax.validation.constraints.NotNull;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+public class Config {
+    public static final String TARGET_ENVIRONMENT = "TARGET_ENVIRONMENT";
+    public static final String PLATFORM = "PLATFORM";
+    public static final String APP_NAME = "APP_NAME";
+    public static final String IS_VISUAL = "IS_VISUAL";
```

Comment:
Should there be a way to add dynamic attributes? That way we support default attributes out of the box, and allow a way to extend for the consumers

Reply:
We can configure any attribute we want in ReportPortalConfigurationManager. It is not just limited to the variables defined here.

Reply:
Let's add this in the README, and also show an example in the utility for this

Reply:
Added the details in README.md file. We have good amount of examples in ReportPortalPropertiesOverloader class for this. All properties under `setSystemAttributes()` method are added like that.

Reply:
Changes done:
1. Added capability to configure dynamic attributes. `setRPAttributesFromSystemVariables` method under `ReportPortalPropertiesOverloader` is taking care of this. Any attribute set at System property or System environment variable with prefix as `RP_` will be set as ReportPortal launch attribute
2. Now, while setting CI related attributes we're checking values configured at System property, System environment and `config.properties` level. `config.properties` takes the least precedence and only if the values is not retrieved from the first two, then only the value from `confg.properties` is taken into account. To be clear, we're not setting the actual values in `config.properties` as these values are dynamic in nature with every pipeline run, and rather are giving the option to consumer to define the key names which their pipeline is using to set those vairables. Also, if the value is not retireved from all 3 levels, then we're setting the value as `not-set`
3. We have updated the README.md explaining all the above changes and how to use them with examples